### PR TITLE
Fixes ticket #66 by implementing a PlonePAS plugin for properties.

### PR DESCRIPTION
**Pull request by pcdummy**
Everything is documented in Ticket #66 

Signed-off-by: Rene Jochum <rene@jochums.at>

### CODE REVIEW

File: src/bda/plone/shop/user/properties.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        country = getProperty('country', u'')
+
+        try:
+            country = country and get_pycountry_name(country) or ''
+        except KeyError:
+            country = ''
+
+        join_list = [street, '{0} {1}'.format(zip_code, city), country]
+
+        # return {
```
Comment (rnixx) [inline src/bda/plone/shop/user/properties.py]: Can you remove the commented return statement please?

File: src/bda/plone/shop/user/userdata.py [hunk 2]
```diff
@@ -166,11 +166,13 @@ class UserDataSchemaAdapter(AccountPanelSchemaAdapter):
 
     @property
     def fullname(self):
+        # TODO: I think this never gets called as there
+        # was an error in the string formatting.
```
Comment (rnixx) [inline src/bda/plone/shop/user/userdata.py]: Have you tried adding a pdb? I'm not sure whether PAS just swallows the resulting AttributeError

Comment (rnixx) [inline src/bda/plone/shop/user/userdata.py]: @thet what was this code good for?

https://github.com/bluedynamics/bda.plone.shop/blame/master/src/bda/plone/shop/user/userdata.py#L167

You implemented exactly the computed properties ``location`` and ``fullname`` in UserDataSchemaAdapter which are now in the custom PAS plugin.

Did the lookup in the related plone views change?

Comment (rnixx) [inline src/bda/plone/shop/user/userdata.py]: Also the objection from @pcdummy is interesting, the code must have always failed

**Comment by rnixx** [comment]
Basically this PR looks good. Anyway I want to know which change initially broke user name displaying just to get sure the custom property sheet is the correct way to fix this problem.

**Comment by rnixx** [comment]
Also it would be interesting whether we can remove ``fullname`` and ``location`` properties from ``UserDataSchemaAdapter``

**Comment by pcdummy** [comment]
I'm pretty sure ``UserDataSchemaAdapter`` can be removed, i didn't remove it as i'm not sure what it is good for.

**Comment by rnixx** [comment]
``UserDataSchemaAdapter`` is needed for the checkout to provide default values in the checkout form IIRC

**Comment by rnixx** [comment]
ah, no, this was ``CheckoutFormMemberPresets``

**Comment by rnixx** [comment]
@pcdummy I think you can comment it out for now. Let's remove it once we know definitely that we don't need it

**Comment by pcdummy** [comment]
I don't see another way to resolve issue #66 for me its the only way to get into getProperty of [PlonePAS](https://github.com/plone/Products.PlonePAS/blob/master/src/Products/PlonePAS/tools/memberdata.py#L313)

**Comment by rnixx** [comment]
@pcdummy The point is it already worked. So something must have been changed in order to break our custom user data and I would really appreciate to know what it was prior to merging.

**Comment by pcdummy** [comment]
@rnixx Do you know up to which version of Plone and bda.plone.shop this worked? Then i might find the change which broke this.

**Comment by thet** [comment]
Not sure if it's worth the effort. IMO more important is, that this approach works both in Plone 4 and Plone 5.

**Comment by pcdummy** [comment]
[This](https://github.com/bluedynamics/bda.plone.shop/commit/8e7c9eaff5e6653bdba56d09db1b294c9080933c#diff-38e932be4d0debc2101ec64e32fc69f6R3) isn't it?

**Comment by pcdummy** [comment]
Just tried it, we need the ``UserDataSchemaAdapater``. But not the properties ``fullname`` and ``location``.

**Comment by rnixx** [comment]
@pcdummy It was not a change in b.p.shop which broke the displaying

**Comment by pcdummy** [comment]
@rnixx and @thet Did it show the username or a fullname then?

**Comment by rnixx** [comment]
@pcdummy the username

**Comment by rnixx** [comment]
@pcdummy probably @thet is right about "worth the effort". I'm going to merge this now. Thank you!

**Comment by pcdummy** [comment]
Yeha!

**Comment by thet** [comment]
@pcdummy did you try this out in Plone 5 also? In Plone 5 there were a lot of changes regarding custom user data - they can even be set TTW.

**Comment by pcdummy** [comment]
The bug itself, yes. The bugfix, no. Will test it right now.

**Comment by pcdummy** [comment]
It works with Plone5 but i found a bug, the new Plugin must be last so this works and it doesn't get sorted in as the last one.

I'll open a new PR and reopen 66.

**Pull request merged**